### ISSUE TITLE
Hi, I have a question about performance on AMD processors 

### ISSUE THREAD

**Issue by ov3rdr1v337** [xdp-project/xdp-tutorial#170]
Well, i'm always running XDP applications on server Intel Gold Processors, and the performance good - up to 125Mpps, but today i got very cool AMD proc - AMD EPYC 7371, install ubuntu serv 18.04(later - 20.04), installed mallanox driver(i have two connectx 5-ver 100GBE meallanox net cards, and two servers - one generator, one receiver) - OFED. Everything is as usual, and i launched pereformance test on FULL DROP XDP DROP filter, and i got... only 13Mpps. And I have no idea why so few?

Here is some tunning tricks that's i used:

sudo mlnx_tune -p HIGH_THROUGHPUT
sudo ethtool -G enp65s0 rx 512 tx 512
sudo ethtool -L enp65s0 combined 8
sudo ethtool --show-priv-flags enp65s0
sudo ethtool --set-priv-flags enp65s0 rx_cqe_compress on

can u help me, pls?

**Comment by roddylab** [comment]
A bit more info about our config: kernel 5.4, we generate traffic from our 2nd machine (TRex) which also has MCX515A-CCAT network card and is connected to 1st machine (AMD one) with Mellanox 100 GbE cable

**Comment by tohojo** [comment]
Missing DDIO? @netoptimizer any thoughts?

**Comment by roddylab** [comment]
> Missing DDIO? @netoptimizer any thoughts?

Yeah, we have thought about that, but is it possible for the performance to drop by over 100Mpps just because we lack DDIO?
Also Cloudflare who use XDP mention that they exchanged their Intel Xeon Platinum 8160 CPU's (2 per unit) to AMD 7442 CPU (1 per unit) and got better results by far.

Thanks for your fast reply btw

**Comment by tohojo** [comment]
roddy <notifications@github.com> writes:

>> Missing DDIO? @netoptimizer any thoughts?
>
> Yeah, we have thought about that, but is it possible for the
> performance to drop by over 100Mpps just because we lack DDIO? Also
> Cloudflare who use XDP mention that they exchanged their Intel Xeon
> Platinum 8160 CPU's (2 per unit) to AMD 7442 CPU (1 per unit) and got
> better results by far.

No, it does seem a bit much. You don't mention CPU affinity tuning, did
you do any? What does `mpstat -P ALL 1` output while you're running your
test? I.e., are you seeing activity on all CPUs?

> Thanks for your fast reply btw

You're welcome :)

**Comment by netoptimizer** [comment]
> roddy <notifications@github.com> writes:
> > Missing DDIO? @netoptimizer any thoughts? 
>
> Yeah, we have thought about that, but is it possible for the performance to drop by over 100Mpps just because we lack DDIO? Also Cloudflare who use XDP mention that they exchanged their Intel Xeon Platinum 8160 CPU's (2 per unit) to AMD 7442 CPU (1 per unit) and got better results by far.
> No, it does seem a bit much. You don't mention CPU affinity tuning, did you do any? What does `mpstat -P ALL 1` output while you're running your test? I.e., are you seeing activity on all CPUs?

I've wanted to play with the AMD EPYC series CPUs for a while now... this seems like a good opportunity to interact on this.
I'm particularly interested in if DDIO / DCA works on AMD. Notice if this tech is not available, then (back before DDIO enabled CPUs) I had software patches to the mlx5 driver, that solved/removed the full cache-miss, via L2-cache prefetching, but we (the kernel) choose not to merge this.

First step to check if any (full) cache-misses occur via below `perf stat` command:

`
perf stat -C insert-cpu-number -e cycles -e  instructions -e cache-references -e cache-misses -e branches:k -e branch-misses:k -r 4 sleep 1
`

Remember to change **insert-cpu-number** to one of the CPUs that is active processing/doing `XDP_DROP`. 
I've also included som other stats, that will tell me other things.
(Please see if you can fit and format results in a readable fashion via this GitHub interface.)

**Comment by netoptimizer** [comment]
> > @roddylab / @tohojo  writes:
> > > Missing DDIO? @netoptimizer any thoughts?
> > 
> 
> First step to check if any (full) cache-misses occur via below `perf stat` command:

Hmm... silence from @ov3rdr1v337 and @roddylab
